core/multigrid/pgm.cpp

```diff
@@ -270,20 +277,29 @@ void Pgm<ValueType, IndexType>::communicate(
         send_agg.get_size(), local_agg.get_const_data(),
         gather_idxs.get_const_data(), send_agg.get_data()));
 
+    // temporary index map that contains no remote connections to map
+    // local indices to global
+    experimental::distributed::index_map<IndexType, GlobalIndexType> imap(
+        exec, coarse_partition, comm.rank(), array<GlobalIndexType>{exec});
+    auto seng_global_agg = imap.map_to_global(
+        send_agg, experimental::distributed::index_space::local);
```

Comment:
If I guess it correctly, you map the local index back to the global index to avoid the collision. 
this is what I want to avoid because it needs twice of memory transferring between nodes.

Reply:
I would guess that the actual volume of the data transfer is negligible compared to the latency cost. Also, this is significantly simpler to what was there before.